simulator/src/main/java/com/hedera/block/simulator/generator/BlockStreamManager.java

```diff
@@ -16,15 +16,31 @@
 
 package com.hedera.block.simulator.generator;
 
+import com.hedera.block.simulator.config.types.GenerationMode;
 import com.hedera.hapi.block.stream.Block;
+import com.hedera.hapi.block.stream.BlockItem;
 
-/** this interface defines the contract for managing the block stream. */
+/** The block stream manager interface. */
 public interface BlockStreamManager {
 
     /**
-     * Gets the next block in the block stream.
+     * Get the generation mode.
      *
-     * @return the next block in the block stream
+     * @return the generation mode
+     */
+    GenerationMode getGenerationMode();
+
+    /**
+     * Get the next block item.
+     *
+     * @return the next block item
+     */
+    BlockItem getNextBlockItem();
+
+    /**
+     * Get the next block.
+     *
+     * @return the next block
      */
     Block getNextBlock();
```

Comment:
Block items should always be bundled within a Block in the correct order. Perhaps we can remove the `getNextBlockItem()` method from the interface and rely only on `getNextBlock()`?